Upgrade untool to 2.0.0-alpha.4 (#971)
xing/hops: README.md

    - [Default settings](#default-settings)
    - [Placeholders](#placeholders)
    - [Environment variables](#environment-variables)

>     - [Controlling what will be exposed on the client](#controlling-what-will-be-exposed-on-the-client)

Reviewer: With the whitelist this is more about `Exposing config values to the client` rather than `Controlling what will be exposed…`, because by default there's nothing exposed besides the `basePath`. So this whole topic isn't about taking care not to expose secrets accidentally anymore, but rather to expose explicitly, what can be exposed.

| `serverDir` | `String` | `node_modules/.cache/untool` | `<rootDir>/dist` | The directory where the generated server middleware will be stored |
| `browsers` | `Array<String>` | `['defaults']` | `['last 1 Chrome versions']` | An array of browserslist queries to specify targets for which to transpile/polyfill (see [`@babel/preset-env`](https://babeljs.io/docs/en/babel-preset-env) for more information) |
| `node` | `String` | `current` | `8.10` | A Node.js version identifier or `current` to specify for which target to transpile/polyfill |

> | `browserWhitelist` | `Object` | `{"basePath":true}` | A map of config keys that should be whitelisted and exposed to the client. Nested paths can be described with dot notation |

Reviewer: Mentioning that those config keys are whitelisted, is meaningless in this context imho.

```diff
- A map of config keys that should be whitelisted and exposed to the client.
+ A map of config keys that should be exposed to the client.
```

`Using` sounds a little better than `with` here imho.

```diff
- Nested paths can be described with dot notation
+ Nested paths can be described using dot notation
```

> **But:** Since Hops renders applications on the server- and client-side, we can reference environment variables in the settings and pass them on to the client and therefore allow users to have environment variables that are evaluated at runtime.

Reviewer: This needs a hint, that exposing those values to the client-side needs an additional effort.

> **Caution:** Be careful not to expose secrets to the client side, because all environment variables that you reference this way in the settings will be visible in the rendered HTML. Read how you can [control what will be exposed on the client](#controlling-what-will-be-exposed-on-the-client).

Reviewer: This whole paragraph can either be removed.

**Tip:** Hops ships with integrated support for [`dotenv`](https://github.com/motdotla/dotenv), which means that it will try to read a `.env` file from your application root directory and load its values as environment variables.

#### Exposing values to the client

> Hops requires you to whitelist specific values from your configuration that will be exposed to the client side. In order to configure which values you need access to on the client side, you can map the configuration keys to boolean values. For nested objects you can use the dot notation:

Reviewer 2: ```suggestion
Hops requires you to whitelist all values from your configuration that needs to be exposed to the client side. In order to do so, you can map the configuration keys to boolean values. For nested objects you can use the dot notation:
```

We could also write `accessible on the client` instead of `exposed to the client`.

**Tip:** Hops ships with integrated support for [`dotenv`](https://github.com/motdotla/dotenv), which means that it will try to read a `.env` file from your application root directory and load its values as environment variables.

#### Controlling what will be exposed on the client

> Hops allows you to whitelist specific values from your configuration that will be exposed to the client side. In order to configure which values you need access to on the client side, you can map the configuration keys to boolean values. For nested objects you can use the dot notation:

Reviewer: Hops doesn't allow, it requires the user to do this additionally.
